cli/command/trust/signer_remove.go

```diff
@@ -0,0 +1,135 @@
+package trust
+
+import (
+	"fmt"
+
+	"os"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+	"github.com/spf13/cobra"
+)
+
+type signerRemoveOptions struct {
+	forceYes bool
+	images   []string
+	signer   string
+}
+
+func newSignerRemoveCommand(dockerCli command.Cli) *cobra.Command {
+	options := signerRemoveOptions{}
+	cmd := &cobra.Command{
+		Use:   "signer-remove NAME IMAGE [IMAGE...]",
+		Short: "Remove a signer",
+		Args:  cli.RequiresMinArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.signer = args[0]
+			options.images = args[1:]
+			return removeSigner(dockerCli, &options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.forceYes, "yes", "y", false, "Answer yes to removing most recent signer (no confirmation)")
+	return cmd
+}
+
+func removeSigner(cli command.Cli, options *signerRemoveOptions) error {
+	if len(options.images) < 1 {
+		return fmt.Errorf("at least one image must be provided")
+	}
+
+	for _, image := range options.images {
+		if err := removeSingleSigner(cli, image, options.signer, options.forceYes); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isLastSigner(role data.Role, notaryRepo *client.NotaryRepository) (bool, error) {
```

Comment:
nit: it'll be easier to test this function if it takes the output of `notaryRepo.ListRoles()` as an arg instead of a direct `NotaryRepository`